waku/node/waku_node.nim

```diff
@@ -389,14 +387,45 @@ proc filterSubscribe*(node: WakuNode, pubsubTopic: PubsubTopic, contentTopics: C
 
       await handler(pubsubTopic, message)
 
-  let subRes = await node.wakuFilterClientLegacy.subscribe(pubsubTopic, contentTopics, handlerWrapper, peer=remotePeer)
-  if subRes.isOk():
-    info "subscribed to topic", pubsubTopic=pubsubTopic, contentTopics=contentTopics
+  var topics: seq[ContentTopic]
+  when contentTopics is seq[ContentTopic]:
+    topics = contentTopics
   else:
-    error "failed filter subscription", error=subRes.error
-    waku_node_errors.inc(labelValues = ["subscribe_filter_failure"])
-
-proc filterUnsubscribe*(node: WakuNode, pubsubTopic: PubsubTopic, contentTopics: ContentTopic|seq[ContentTopic],
+    topics = @[contentTopics]
+
+  var topicMap = initTable[PubsubTopic, seq[ContentTopic]]()
+  for contentTopic in topics:
+    let res = parseSharding(pubsubTopic, contentTopic)
+    
+    let (pubsub, content) =
+      if res.isErr():
+        error "parsing error", error = res.error
+        return
+      else: res.get()
+    
+    if not topicMap.hasKey(pubsub):
+      topicMap[pubsub] = @[]
+    
+    topicMap[pubsub].add(content)
```

Comment:
Can we extract all this logic into a simple helper function in topics/sharding.nim (could overload `parseSharding`) that takes a `pubsubTopic` option and sequence of content topics and returns a map from pubsub topic to content topics. Will prevent waku_node.nim from bloating even further and can be reused for the same logic in the `unsubscribe` below.

Reply:
that would be better yes